collectors/test-results/jira-xray/src/main/java/com/capitalone/dashboard/api/domain/TestRun.java

```diff
@@ -0,0 +1,218 @@
+package com.capitalone.dashboard.api.domain;
+
+import com.atlassian.jira.rest.client.api.domain.BasicIssue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * This class will get the details from a Test Run
+ */
+public class TestRun extends BasicIssue implements Versionable<TestRun> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TestRun.class);
+    private TestRun oldVersion;
+    private int version=0;
+
+    private Status status;
+    private String executedBy;
+    private String assignee;
+    private Date startedOn;
+    private Date finishedOn;
+    private Iterable<Defect> defects;
+    private Iterable<Evidence> evidences;
+    private Comment comment;
+
+    private Iterable<TestStep> steps;
+    private Iterable<Example> examples;
+
+    //TODO: Need to be implemented scenario &
+
+    private String testExecKey;
+
+    public TestRun(URI self,String key,Long id){
+        super(self,key,id);
+    }
+
+    public TestRun(URI self, String key, Long id, Status status, Date startedOn, Date finishedOn, String assignee, String executedBy, Iterable<TestStep>steps) {
+        super(self, key, id);
+        this.status=status;
+        this.assignee=assignee;
+        this.executedBy=executedBy;
+        this.startedOn=startedOn;
+        this.finishedOn=finishedOn;
+        this.steps=steps;
+        try {
+            this.oldVersion= cloneTestRun();
+        } catch (CloneNotSupportedException e) {
+            LOGGER.error("Clone is not supported for the TestRun: " + e);
+        }
+    }
+
+    // TODO: ADD CLONE TO ARRAYS USED IN THE TEST RUN
+    public final TestRun cloneTestRun() throws CloneNotSupportedException {
+        TestRun myTestRun= null;
+            myTestRun = new TestRun(super.getSelf(),super.getKey(),super.getId());
+            if(this.executedBy!=null)
```

Comment:
@kumarvarun1252  use Optional for null checks

Reply:
Optional used